backend/DevUp/tests/DevUp.Api.Tests.Integration/V1/Identity/IdentityControllerTests.cs

```diff
@@ -61,28 +59,48 @@ public async Task Register_WhenGivenInvalidRequest_ReturnsBadRequest()
+        public async Task Register_WhenGivenDuplicateUsername_ReturnsBadRequestWithDetailedError()
         {
-            var faker = new IdentityFaker();
-            var resultReg = await _apiClient.PostAsJsonAsync(Route.Api.V1.Identity.Register, faker.RegisterUserRequest);
+            var otherFaker = new IdentityFaker();
+            var duplicatedUsernameRegisterRequest = new RegisterUserRequest()
+            {
+                Username = _faker.RegisterUserRequest.Username,
+                Password = otherFaker.RegisterUserRequest.Password,
+                Device = otherFaker.RegisterUserRequest.Device,
+            };
 
-            var result = await _apiClient.PostAsJsonAsync(Route.Api.V1.Identity.Login, faker.LoginUserRequest);
-            var response = await result.Content.ReadFromJsonAsync<IdentityResponse>();
+            var nonDuplicatedResult = await _apiClient.PostAsJsonAsync(Route.Api.V1.Identity.Register, _faker.RegisterUserRequest);
+            nonDuplicatedResult.Should().HaveStatusCode(HttpStatusCode.OK);
+
+            var duplicatedResult = await _apiClient.PostAsJsonAsync(Route.Api.V1.Identity.Register, duplicatedUsernameRegisterRequest);
+            duplicatedResult.Should().HaveStatusCode(HttpStatusCode.BadRequest);
+
+            var response = await duplicatedResult.Content.ReadFromJsonAsync<ErrorResponse>();
+            response!.Errors.Should().ContainSingle(e => e == "User with this username already exist.");
+        }
+
+        [Fact]
+        public async Task Login_WhenGivenValidCredentialsOfAnExistingUser_ReturnsTokenPair()
+        {
+            var resultReg = await _apiClient.PostAsJsonAsync(Route.Api.V1.Identity.Register, _faker.RegisterUserRequest);
 
+            var result = await _apiClient.PostAsJsonAsync(Route.Api.V1.Identity.Login, _faker.LoginUserRequest);
             result.Should().HaveStatusCode(HttpStatusCode.OK);
+
+            var response = await result.Content.ReadFromJsonAsync<IdentityResponse>();
             response!.Token.Should().NotBeEmpty();
             response.RefreshToken.Should().NotBeEmpty();
         }
 
         [Fact]
-        public async Task Login_WhenGivenInvalidCredentials_ReturnsBadRequest()
+        public async Task Login_WhenGivenInvalidCredentials_ReturnsBadRequestWithGenericMessage()
         {
             // try to login without registering
             var result = await _apiClient.PostAsJsonAsync(Route.Api.V1.Identity.Login, _faker.LoginUserRequest);
-            var response = await result.Content.ReadFromJsonAsync<ErrorResponse>();
-
             result.Should().HaveStatusCode(HttpStatusCode.BadRequest);
-            response!.Errors.Should().ContainSingle(e => e == LoginException.InvalidUsernameMessage);
+
+            var response = await result.Content.ReadFromJsonAsync<ErrorResponse>();
+            response!.Errors.Should().ContainSingle(e => e == "Invalid request.");
```

Comment:
I'm not sure if that's the proper way of doing this. As a user I'm trying to log in, and I'm given `Invalid request` response, suggesting the request wasn't understood or correctly processed by servuur. I'd suggest using 401 (Unauthorized) code, with message: "Invalid credentials" - it's generic, doesn't provide information whether login, password or both are invalid.

Reply:
I think that code 400 is correct one here (unless we want to go deeper and use 422). 401 doesn't make much sense in this case, in my understanding it should be used when we want to access something and we're not authorized _yet_. In this particular case, login endpoint is not something that _requires_ user to be authorized, it's an endpoint that _grants_ authorization, therefore I'd stick to 400 (or 422, but that'd be for another pull request as change scope would have to be much greater).

Reply:
Checked some sources, and couldn't find one answer - hence it seems like it's not standardized. We can leave it as is.